de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/window/api/impl/ADUIElementFieldDAO.java

```diff
@@ -0,0 +1,137 @@
+package org.adempiere.ad.window.api.impl;
+
+import static org.adempiere.model.InterfaceWrapperHelper.copy;
+import static org.adempiere.model.InterfaceWrapperHelper.newInstance;
+import static org.adempiere.model.InterfaceWrapperHelper.save;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.ad.trx.api.ITrx;
+import org.adempiere.ad.window.api.IADFieldDAO;
+import org.adempiere.ad.window.api.IADUIElementFieldDAO;
+import org.adempiere.util.Services;
+import org.adempiere.util.proxy.Cached;
+import org.compiere.model.I_AD_Field;
+import org.compiere.model.I_AD_Tab;
+import org.compiere.model.I_AD_UI_Column;
+import org.compiere.model.I_AD_UI_Element;
+import org.compiere.model.I_AD_UI_ElementField;
+import org.compiere.model.I_AD_UI_ElementGroup;
+import org.compiere.model.I_AD_UI_Section;
+import org.compiere.util.Env;
+import org.slf4j.Logger;
+
+import de.metas.adempiere.util.CacheCtx;
+import de.metas.adempiere.util.CacheTrx;
+import de.metas.logging.LogManager;
+
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+public class ADUIElementFieldDAO implements IADUIElementFieldDAO
+{
+	private static final transient Logger logger = LogManager.getLogger(ADUIElementFieldDAO.class);
+
+	@Override
+	public void copyUIElementFields(final I_AD_UI_Element targetUIElement, final I_AD_UI_Element sourceUIElement)
+	{
+		final Map<Integer, I_AD_UI_ElementField> existingTargetUIElementFields = retrieveUIElementFields(Env.getCtx(), targetUIElement.getAD_UI_Element_ID(), ITrx.TRXNAME_ThreadInherited);
+		final Collection<I_AD_UI_ElementField> sourceUIElementFields = retrieveUIElementFields(Env.getCtx(), sourceUIElement.getAD_UI_Element_ID(), ITrx.TRXNAME_ThreadInherited).values();
+
+		sourceUIElementFields.stream()
```

Comment:
instead of `stream().foreach()` please use the good old for-loop. It's much easier to debug

Reply:
Done